### DISCUSSION TITLE
Converting IFC to OBJ While Preserving the Original Object Structure

### DISCUSSION THREAD

**Discussion by radeksmejky9** [IfcOpenShell/IfcOpenShell#5813] [Q&A] [answered]
Hello, I've been working on an AR BIM viewer app, and I've run into a challenge while preparing the API for file conversion on my web server. I can successfully convert the file itself, but I'm not sure how to preserve the structure of the objects. I need to identify the object types by their names using regex. Can you help me figure out how to do this?

**Comment by aothms** [comment, marked as answer]
> the structure of the objects

First of all, keep in mind that OBJ might be a bad fit for what you're trying to do. It doesn't have per-object matrices or meta-data. Something like Collada or glTF would be a much better fit.

> I need to identify the object types by their names using regex

Can you more precisely state what you want to achieve, where is this regex matching intended to take place. IfcConvert has a `--use-element-names` output option, but use with care. Names in IFC are not necessarily unique.

**Reply by radeksmejky9** [reply]
Thanks for such a quick reply! I need an OBJ so I can import it into Unity remotely. After importing, I use regex on the object hierarchy (specifically object names) to identify objects that are named like Cold Water, Hot Water, Waste, Electricity, etc., and apply materials accordingly. However, even with the `--use-element-names` option, the output ends up as a single object.

Is there any way of achieving the outcome on the screen below?

### This is what I want to achieve
![This is what I want to achieve](https://github.com/user-attachments/assets/93f2466c-30ca-4897-9173-f4ec9d5f53f2)

### Current result
![Current result](https://github.com/user-attachments/assets/daa8d768-ee58-43bd-95ea-f5c098014339)

**Reply by aothms** [reply]
Blender has an option on what to split. OBJ is not a well-behaving standard, these kind of differences are to be expected.

![afbeelding](https://github.com/user-attachments/assets/2516fb70-101f-4cec-8052-0809506f4e3e)

**Reply by radeksmejky9** [reply]
I decided to swap over to Collada and suddenly the structure is exactly like I wanted it in the beggining. Thanks a lot.
